polygon/heimdall/simulator/bor_events.go

```diff
@@ -0,0 +1,45 @@
+package simulator
+
+import (
+	"context"
+	"time"
+
+	"github.com/ledgerwatch/erigon-lib/downloader/snaptype"
+	"github.com/ledgerwatch/erigon/polygon/heimdall"
+	"github.com/ledgerwatch/erigon/turbo/snapshotsync/freezeblocks"
+)
+
+func continueLoop(events []*heimdall.EventRecordWithTime, to time.Time, limit int) bool {
+	if len(events) == 0 {
+		return true
+	}
+	if len(events) == limit {
+		return false
+	}
+
+	last := events[len(events)-1]
+	return last.Time.Before(to)
+}
+
+func (h *HeimdallSimulator) getEvents(ctx context.Context, fromId uint64, to time.Time, limit int) ([]*heimdall.EventRecordWithTime, error) {
```

Comment:
please keep all receivers in the same file - code navigation and discovery gets difficult otherwise

Reply:
you can just move this logic inside FetchStateSyncEvents - no need for a separate func